test-suite/js/src/test/scala/org/scalajs/testsuite/javalib/lang/SystemJSTest.scala

```diff
@@ -57,4 +59,56 @@ class SystemJSTest {
       assertEquals(x1FirstHash, System.identityHashCode(x1))
     }
   }
+
+  @Test def systemProperties(): Unit = {
+    def get(key: String): String = java.lang.System.getProperty(key)
+
+    // Defined in System.scala
+    assertEquals("1.8", get("java.version"))
+    assertEquals("1.8", get("java.vm.specification.version"))
+    assertEquals("Oracle Corporation", get("java.vm.specification.vendor"))
+    assertEquals("Scala.js", get("java.vm.name"))
+    assertEquals("1.8", get("java.specification.version"))
+    assertEquals("/", get("file.separator"))
+    assertEquals(":", get("path.separator"))
+    assertEquals("\n", get("line.separator"))
+
+    // Added via __ScalaJSEnv
+
+    assertNotNull(get("java.vm.version"))
+    assertNotNull(get("os.name"))
+    assertNotNull(get("os.version"))
+    assertNotNull(get("os.arch"))
+    assertNotNull(get("scalajs.compliant-moduleinit"))
+    assertNotNull(get("scalajs.compliant-asinstanceofs"))
+    assertNotNull(get("scalajs.production-mode"))
+
+    val inNode = get("scalajs.node") == "true"
+    val inPhantom = get("scalajs.phantom") == "true"
+    val inRhino = get("scalajs.rhino") == "true"
+    val hasTypedArrays = get("scalajs.typedarray") == "true"
```

Comment:
`hasTypedArrays` is misleading. The property `typedarray` is enabled independently of whether the VM has typed arrays. It is set when we want to test typed array tests. It is possible that the VM supports typed arrays but `scalajs.typedarray` is `false`.